extended/src/main/java/io/kubernetes/client/extended/event/legacy/LegacyEventBroadcaster.java

```diff
@@ -0,0 +1,160 @@
+import java.util.Optional;
+import java.util.concurrent.*;
+import org.apache.commons.lang3.tuple.MutablePair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LegacyEventBroadcaster implements EventBroadcaster {
+
+  private static final Logger logger = LoggerFactory.getLogger(LegacyEventBroadcaster.class);
+
+  public LegacyEventBroadcaster(CoreV1Api coreV1Api) {
+    this(
+        new EventCorrelator(),
+        new EventSink() {
+          @Override
+          public V1Event create(V1Event event) throws ApiException {
+            return coreV1Api.createNamespacedEvent(
+                event.getMetadata().getNamespace(), event, null, null, null);
+          }
+
+          @Override
+          public V1Event update(V1Event event) throws ApiException {
+            return coreV1Api.replaceNamespacedEvent(
+                event.getMetadata().getName(),
+                event.getMetadata().getNamespace(),
+                event,
+                null,
+                null,
+                null);
+          }
+
+          @Override
+          public V1Event patch(V1Event event, V1Patch patch) throws ApiException {
+            return PatchUtils.patch(
+                V1Event.class,
+                () ->
+                    coreV1Api.patchNamespacedEventCall(
+                        event.getMetadata().getName(),
+                        event.getMetadata().getNamespace(),
+                        patch,
+                        null,
+                        null,
+                        null,
+                        null,
+                        null),
+                V1Patch.PATCH_FORMAT_STRATEGIC_MERGE_PATCH,
+                coreV1Api.getApiClient());
+          }
+        });
+  }
+
+  public LegacyEventBroadcaster(EventCorrelator correlator, EventSink sink) {
+    this.pendingEventQueue = new LinkedBlockingQueue<>();
+    this.eventProcessingWorker = Executors.newSingleThreadExecutor();
+    this.eventCorrelator = correlator;
+    this.sleepDuration = Duration.ofSeconds(10);
+    this.eventSink = sink;
+  }
+
+  private BlockingQueue<V1Event> pendingEventQueue;
+  private ExecutorService eventProcessingWorker;
+  private EventSink eventSink;
+  private EventCorrelator eventCorrelator;
+  private Duration sleepDuration;
+  private boolean shuttingDown;
+
+  @Override
+  public EventRecorder newRecorder(V1EventSource eventSource) {
+    return new ObjectReferenceResolvingEventRecorder(this.pendingEventQueue);
+  }
+
+  @Override
+  public void startRecording() {
+    this.eventProcessingWorker.submit(
+        () -> {
+          while (true) {
+            try {
+              V1Event event = pendingEventQueue.poll(100, TimeUnit.MILLISECONDS);
+              if (event != null) {
+                recordToSink(event);
+              }
+            } catch (InterruptedException e) {
+              logger.info("shutdown signaled");
+            }
+            if (this.shuttingDown) {
```

Comment:
why not `while (!this.shuttingDown) { ...` instead of this break out of the loop here. I think this is overly complicated.

Reply:
good point, much cleaner